app/frontend/actions.ts

```diff
@@ -603,17 +603,24 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
 
   const prepareTxPlan = async (args) => {
     const state = getState()
-    const plan: any = await wallet.getAccount(state.sourceAccountIndex).getTxPlan(args)
-    // FIXME: this cant be any
-    if (plan.error) {
-      stopLoadingAction(state, {})
-      resetDelegationWithoutHash(state)
-      setState({
-        calculatingDelegationFee: false,
-        calculatingFee: false,
-      })
+    try {
+      const plan: any = await wallet.getAccount(state.sourceAccountIndex).getTxPlan(args)
+      // FIXME: this cant be any
+      if (plan.error) {
+        stopLoadingAction(state, {})
+        resetDelegationWithoutHash(state)
+        setState({
+          calculatingDelegationFee: false,
+          calculatingFee: false,
+        })
+      }
+      return plan
+    } catch (e) {
```

Comment:
I would restrict this to network errors if that's our concern to not silence other exceptions that may be e.g. related to some bug or other kind of malfuncion

Reply:
You are right, I didnt get to respond to the previous comment but our concern are the network errors indeed. All the other errors that might occur would still be displayed somewhere but you are right that they wouldn't be "unexpected". I ll add  
```
If (e.name !== 'NetworkError` || e.name !== 'ServerError') throw e
```

Reply:
I definitely plan to refactor this before the multi-asset so this whole tx plan creation works a bit better :D